### PR TITLE
Extend the ClusterAdmissionPolicy resource to allow binding to a Policy Server

### PR DESCRIPTION
**Pull request by raulcabello**
Fixes kubewarden/kubewarden-controller#49

### CODE REVIEW

File: PROJECT [hunk 1]
```diff
@@ -8,18 +8,18 @@ resources:
...
     namespaced: false
+  controller: true
   domain: kubewarden.io
   group: policies
-  kind: ClusterAdmissionPolicy
-  path: github.com/kubewarden/kubewarden-controller/apis/policies/v1alpha2
-  version: v1alpha2
+  kind: PolicyServer
+  path: github.com/kubewarden/kubewarden-controller/apis/policies/v1alpha3
+  version: v1alpha3
```
Comment (flavio) [inline PROJECT:16]: I know we discussed that yesterday, but do we have to update the api version? Given the new field is optional, can't we just add it to the existing v1alpha2 policy?

Moreover, if we are forced to add a new version of the object, the old one must stay around for some time for compatibility reasons.

File: apis/policies/v1alpha3/clusteradmissionpolicy_types.go [hunk 2]
```diff
@@ -23,6 +23,11 @@ import (
 
 // ClusterAdmissionPolicySpec defines the desired state of ClusterAdmissionPolicy
 type ClusterAdmissionPolicySpec struct {
+	// PolicyServer identifies an existing PolicyServer resource.
+	// +kubebuilder:default:=default
```
Comment (flavio) [inline apis/policies/v1alpha3/clusteradmissionpolicy_types.go:27]: Nice, we don't have to do anything fancy to have `default` being written into this field :partying_face:

File: config/crd/bases/policies.kubewarden.io_clusteradmissionpolicies.yaml [hunk 3]
```diff
@@ -194,6 +194,10 @@ spec:
                       are ANDed.
                     type: object
                 type: object
+              policyServer:
+                default: default
+                description: PolicyServer identifies an existing PolicyServer resource.
+                type: string
```
Comment (ereslibre) [inline config/crd/bases/policies.kubewarden.io_clusteradmissionpolicies.yaml:200]: So this is what `+kubebuilder:default:=default` is generating, nice. I wasn't sure if `+kubebuilder:default:=default` would require a webhook, but is really nice it doesn't and uses this API machinery feature :)

**Review by ereslibre — APPROVED** [review]
LGTM

**Review by flavio — CHANGES_REQUESTED** [review]
I just skimmed through the PR, there are some comments at the beginning that influence the whole change

**Review by ereslibre — CHANGES_REQUESTED** [review]
In this case, since this is a change that we didn't merge at all, I'd squash all commits into one, to keep history meaningful.

**Comment by raulcabello** [comment]
@ereslibre yes, you are right. I was going to squash the 3 commits before merging in GH

**Review by flavio — APPROVED** [review]
LGTM, in this case I would squash the commits together

**Comment by raulcabello** [comment]
done, thanks!

**Pull request merged**